pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix

```diff
@@ -120,6 +120,8 @@ self: super:
 
   cryptography = super.cryptography.overridePythonAttrs (
     old: {
+      nativeBuildInputs = old.nativeBuildInputs or []
+        ++ stdenv.lib.optional (stdenv.buildPlatform != stdenv.hostPlatform) self.python.pythonForBuild.pkgs.cffi;
```

Comment:
this is fixed on staging-next

Reply:
I based this on staging since this depends on the recent fixes merged there, but these overrides are poetry specific the standard cryptography worked IIRC.